shm/ship.cpp

```diff
@@ -0,0 +1,17 @@
+#include "ship.hpp"
+
+Ship &Ship::operator-=(size_t num)
+{
+    crew_ -= num;
+    return *this;
+}
+
+Ship &Ship::operator+=(size_t num)
+{
+    crew_ += num;
+    return *this;
+}
```

Comment:
What happened when we reach the limit of capacity or drop below 0? You need to handle error cases. Maybe throw exception, maybe return `std::nullopt` or `nullptr` maybe set some specific value like `std::numeric_limits::min/max`.